uk.co.compucorp.civicrm.hrsampledata/CRM/HRSampleData/CSVHandler.php

```diff
@@ -86,6 +91,24 @@ protected function callAPI($entity, $action, $params) {
   }
 
   /**
+   * Deletes entity record from the database based on
+   * search criteria
+   *
+   * @param string $entity
+   * @param array $searchParams
+   */
+  protected function deleteRecord($entity, $searchParams) {
```

Comment:
I liked the idea of trying to make this a more generic thing and renaming the class to CSVHandler and the import to iterate, but things got a little confusing when, for example, the CSVHandler has methods like `deleteRecord()` (which is specific for cleaners) and `callAPI()`.

I think the [Visitor pattern](https://en.wikipedia.org/wiki/Visitor_pattern) would be a perfect fit for this as it solves exactly the problem you tackling here. I created a small gist to show how it could be implemented here: https://gist.github.com/davialexandre/6df1ac26450563cf59ac294a9bc5987e

And this is a diagram of how the classes would work together:
![captura de tela 2016-10-31 as 11 31 29](https://cloud.githubusercontent.com/assets/388373/19855806/9b6cf8a6-9f5d-11e6-9d2f-4142563a0ddc.png)

I just don't know how things are in terms of time for this ticket. If it's something really urgent, then it's ok to leave it as it is now, otherwise It would definitely be worth to improve this and separate the responsibilities in a better way